meta-psa/README.md

```diff
@@ -65,16 +84,181 @@ and processed by the Linux kernel recipe used to build the kernel. For example,
+
+Once installed in the keyring, the FEKEK can be referenced with its signature (e.g.
+`b89bed87989fcf3d`). The following command mounts the encrypted files storage directory
+`/home/root/.secret` to make the decrypted files visible at the `/home/root/secret`
+mount point:
+
+    mount -t ecryptfs /home/root/.secret /home/root/secret -o \
+    ecryptfs_sig=b89bed87989fcf3d,ecryptfs_fnek_sig=b89bed87989fcf3d,\
+    ecryptfs_cipher=aes,ecryptfs_key_bytes=16,\
+    key=passphrase:passphrase_passwd_file=/home/root/.ecryptfs/passphrase.txt,\
+    ecryptfs_passthrough=n,ecryptfs_unlink_sigs,no_sig_cache
+
+### 4.3.2 Security concerns
+The top-level passphrase used to wrap the FEKEK is stored on the system under user account protection.
+This is discussed further in the next section.
+
+## <a name="section-4-4"></a> 4.4 Securing the eCryptfs top-level passphrase
+The previous sections explained that the top-level eCryptfs passphrase is used to decrypt the "wrapped FEKEK" and mount the encrypted storage.
+However, this passphrase is not held securely in the example integration provided. This section provides guidance on how the passphrase can be secured using platform specific hardware services.
+
+### 4.4.1 Securing the eCryptfs passphrase
+In order to store the top-level passphrase securely, the following requirements must be met:
+- The top-level passphrase must be stored securely in the system using hardware enabled security features (e.g. TrustZone).
+- The passphrase may be regenerated when required e.g. using a key generation algorithm to derive a new key from a hardware unique key.
+- The top-level passphrase must be unique to the device, but the same on each system boot.
+- The top-level passphrase must be retrieved (or regenerated) on each boot and populated into the kernel key retention service (keyring).
+  This operation must be performed in a secure manner.
+- eCryptfs must recover the top-level passphrase from the keyring and mount the encrypted storage directory. Once
+  mounted, the top-level passphrase should be removed from the keyring and discarded.
+
+Some IoT devices will include a TPM (Trusted Platform Module) which solves the passphrase storage problem by providing secure
+storage hardware. A hardware TPM typically provides several other security services
+including secure storage of keys and certificates, key provisioning, a secure identity, attestation support
+and anti-rollback protection. Alternatively, a lower-cost firmware TPM using an OPTEE (TrustZone) trusted application
+could be implemented to provide equivalent APIs to those offered by hardware TPMs.
+
+Some IoT devices will include a Secure Element (SE) accessible vai a cryptographic API (e.g. the PSA Crypto API).
+In this case, the SE key storage services are used to store the top level passphrase.
+
+In summary, there are 3 principal methods for securing top-level eCryptfs passphrase:
```

Comment:
```suggestion
There are 3 principal methods for securing top-level eCryptfs passphrase:
```
Drop the "In summary" as you never mention the HUK stuff above so it seems a bit strange as part of a "summary".